### ISSUE TITLE
incomplete refactoring of cursor class hierarchy 

### ISSUE THREAD

**Issue by JackLiJXL** [ubc-systopia/margraphita#65]
currently get {in, out, node, edge} cursor methods are not present in GraphBase. should be included in GraphBase so users can get cursor without knowing representation used for graph

**Comment by apricot2012** [comment]
I think the hierarchy for cursors is actually declared in common.h -- separate from the main graph.h; each of the {in, out, node, edge} cursor constructors return a their parent type so I think we can use them without knowing the representation

**Comment by artorhem** [comment]
yes, we need a pure virtual function in GraphBase to return iters. Sure the graph impls all have it, but if the GraphBase class doesn't have these methods, we can't use a GraphBase pointer to call get_{node, in, out, edge}_iter() methods.
